pydantic/_internal/_typing_extra.py

```diff
@@ -167,9 +167,22 @@ def is_finalvar(ann_type: Any) -> bool:
     return _check_finalvar(ann_type) or _check_finalvar(get_origin(ann_type))
 
 
+_DEFAULT_GLOBALS = [
+    '__name__',
+    '__doc__',
+    '__package__',
+    '__loader__',
+    '__spec__',
+    '__annotations__',
+    '__builtins__',
+    '__file__',
+    '__cached__',
+]
+
+
 def _remove_default_globals_from_ns(namespace: dict[str, Any]) -> dict[str, Any]:
     """Remove default globals like __name__, __doc__, etc that aren't needed for type evaluation."""
-    return {k: v for k, v in namespace.items() if not k.startswith(('__', '@'))}
+    return {k: v for k, v in namespace.items() if not k.startswith('@') and k not in _DEFAULT_GLOBALS}
```

Comment:
I'm wondering, `@` was taken from pytest-examples but is it really possible to have something starting with that in the namespace? `@.*` isn't a valid identifier

Reply:
Good question, yeah, I saw lots of vars starting with `@` in the module namespace when I was running `pytest`, but perhaps that's not important for real applications. I guess I'm inclined to keep it bc it seems like the performance drawback is negligible, and it ensures the namespace is pretty clean.